sdk/inc/azure/iot/az_iot_adu_client.h

```diff
@@ -455,6 +459,15 @@ typedef struct
  */
 AZ_NODISCARD az_iot_adu_client_options az_iot_adu_client_options_default();
 
+/**
+ * @brief Gets the default #az_iot_adu_client_device_properties.
+ * @details Call this to obtain an initialized #az_iot_adu_client_device_properties structure that
+ * can be afterwards modified and passed to necessary APIs.
+ *
+ * @return #az_iot_adu_client_device_properties.
+ */
+AZ_NODISCARD az_iot_adu_client_device_properties az_iot_adu_client_device_properties_default();
```

Comment:
Nit - since this is not an "options" it should be named `az_iot_adu_client_device_properties_init` similar to `az_iot_message_properties_init`.

Reply:
I think here it should be `_default()`. The `_init()` is used to initialize a struct with parameters/values which from then on become private fields of the initialized struct. The interactions on that initialized struct are more or less as though it were an object.

Here, we're giving people a preconfigured default struct which they then directly act on. Thoughts?